VMBackup/main/parameterparser.py

```diff
@@ -102,3 +106,20 @@ def __init__(self, protected_settings, public_settings, backup_logger):
         except Exception as e:
             errorMsg = "Exception occurred while populating includeLunList, Exception: %s" % (str(e))
             backup_logger.log(errorMsg, True)
+        
+        if(self.dynamicConfigsFromCRP != None):
+            try:
+                dynamicConfigsFromCRP_str = json.dumps(self.dynamicConfigsFromCRP)
+                listOfSettings = json.loads(dynamicConfigsFromCRP_str)
+                for dictionary in listOfSettings:
+                    self.settings[dictionary['Key']] = dictionary['Value']
+                backup_logger.log("settings received " + str(self.settings), True)
+                for flag in self.settings.keys():
+                    if(flag not in self.wellKnownSettingFlags):
+                        backup_logger.log("The received " + str(flag) + " is not an expected setting name.", True)
+                    else:
+                        self.wellKnownSettingFlags[flag] = self.settings[flag]
+                backup_logger.log("settings to be sent " + str(self.wellKnownSettingFlags), True)
```

Comment:
`backup_logger.log("settings received " + dynamicConfigsFromCRP_str , True)`
`for config in self.dynamicConfigsFromCRP:`
`    if config['Key'] not in self.wellKnownSettingFlags:`
`        backup_logger.log("The received " + str(config['Key']) + " is not an expected setting name.", True)`
`    else:`
`        self.settings[config['Key']] = config['Value']`
`backup_logger.log("settings to be sent " + str(self.wellKnownSettingFlags), True)`

Treat this as a suggestion if other reviewers OK. Advantage of this:
1. One less for loop
2. One less object. In your implementation, we have 3 objects.